10-habits/progress/internal/repository/tick.go

```diff
@@ -0,0 +1,60 @@
+package repository
+
+import (
+	"context"
+	"log"
+
+	"learngo-pockets/habits/internal/habit"
+	"learngo-pockets/habits/internal/tick"
+)
+
+// ticksPerWeek stores all the ticks per number of weeks.
+type ticksPerWeek map[tick.ISOWeek][]tick.Tick
+
+// TickRepository holds all the current habits with their associated events.
+type TickRepository struct {
+	storage map[habit.ID]ticksPerWeek
+}
+
+// NewTickRepository creates an empty tick repository.
+func NewTickRepository() *TickRepository {
+
+	return &TickRepository{
+		storage: make(map[habit.ID]ticksPerWeek),
+	}
+}
+
+// Add inserts a new event for a habit in memory.
+func (tr *TickRepository) Add(_ context.Context, id habit.ID, t tick.Tick, w tick.ISOWeek) error {
```

Comment:
```suggestion
func (tr *TickRepository) Add(_ context.Context, id habit.ID, w isoweek.ISO8601) error {
```

Reply:
I kept the t as `time.Time` to store the timestamp anyway.

Reply:
But then there is no need for isoweek outside of the repo layer - that's how things are organised _there_.